### ISSUE TITLE
Incompatible With PNPM Workspace

### ISSUE THREAD

**Issue by hyrious** [hyrious/dts#1]
TypeScript has 2 different types: the ones with entities (like `class` `enum`) and the ones without (like `interface`). The previous types _do not merge_. You might have seen this kind of errors:

```
Type 'import("path-1").A' is not assignable to type 'import("path-2").A'.
```

In most of the cases, this is because TypeScript finds 2 definitions of the same-name type in 2 different places. And because `class`es cannot be merged, these 2 definitions are identified as incompatible. It turns out that pnpm has a safe mode to link dependencies to make it impossible to make mistakes to reference an indirect dependency from some package.

```
bar/node_modules/
	foo/ ~> node_modules/.pnpm/foo+space1/node_modules/foo/
buzz/node_modules/
	foo/ ~> node_modules/.pnpm/foo+space2/node_modules/foo/
```

Although the `foo` in `bar` and `buzz` may point to the same `foo` package in the workspace, pnpm still needs to make sure they access to different dependencies in different modules' environment. That will likely to cause the above TypeScript error. While in npm/yarn workspaces, these dependencies are hoisted to the same place so it is less likely to break.

In summary, pnpm workspaces and the `pnpm link` feature may be not friendly to TypeScript.
